### PR TITLE
Gradient matching (Witches' Brew) for Tensorflow

### LINKED ISSUE #1386: Implement Witches’ brew  poisoning attack
**Is your feature request related to a problem? Please describe.**
Implement poisoning attack: 
`Witches' Brew: Industrial Scale Data Poisoning via Gradient Matching` 
https://arxiv.org/abs/2009.02276 

**Describe the solution you'd like**

- Jupyter notebook illustrating the use of the attack 
- Example folder 
- Implementation for Pytorch and Keras framework.

**Describe alternatives you've considered**
NA

**Additional context**
NA

### PR DESCRIPTION
**Pull request by TS-Lee**
# Description

This branch adds Gradient Matching (Witches' Brew) poisoning attack. This attack computes the clean label noise by setting the training gradient of poisoned samples towards the poisoning target class. This PR is for a TF version. The current commit is work in progress, and the follow up will come with a test and an example notebook.

Fixes #1386 

## Type of change

Please check all relevant options.

- [ ] Improvement (non-breaking)
- [ ] Bug fix (non-breaking)
- [x] New feature (non-breaking)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# Testing

Please describe the tests that you ran to verify your changes. Consider listing any relevant details of your test configuration.

- [ ] Test A
- [ ] Test B

**Test Configuration**:
- OS: Ubuntu 18.02
- Python version: 3.8
- ART version or commit number: dev_1.9.0
- TensorFlow / Keras / PyTorch / MXNet version: Tensorflow 2.4.1.

# Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by TS-Lee** [comment]
The target branch is wrong in this PR.. I created a new one: https://github.com/Trusted-AI/adversarial-robustness-toolbox/pull/1587#issue-1167461673
